src/Bedrock.Framework.Experimental/Protocols/Kafka/KafkaMessageReader.cs

```diff
@@ -0,0 +1,67 @@
+﻿#nullable enable
+
+using Bedrock.Framework.Experimental.Protocols.Kafka.Messages.Responses;
+using Bedrock.Framework.Experimental.Protocols.Kafka.Services;
+using Bedrock.Framework.Protocols;
+using System;
+using System.Buffers;
+using System.Diagnostics;
+
+namespace Bedrock.Framework.Experimental.Protocols.Kafka
+{
+    public class KafkaMessageReader : IMessageReader<KafkaResponse>
+    {
+        private readonly IMessageCorrelator correlations;
+
+        public KafkaMessageReader(IMessageCorrelator messageCorrelator)
+        {
+            this.correlations = messageCorrelator;
+        }
+
+        public bool TryParseMessage(in ReadOnlySequence<byte> input, ref SequencePosition consumed, ref SequencePosition examined, out KafkaResponse message)
+        {
+            var reader = new SequenceReader<byte>(input);
+            message = NullResponse.Instance;
```

Comment:
I think just `message = default` is the norm?

Reply:
I've been seeing the Null object pattern a lot more now that nullable is in the language.
I've found it also helps track down when something ends up being a true 'null' after instantiation, as well as for testing purposes.

Reply:
sure, whatever makes you happy 😄 
My theory has always been, if Try__ returns false, never use the out, it is basically "undefined".
But I get your point, given me something to think about